Chat changes

GuilhermeFaga opened the pull request:
- Created a Chat class and gathered chat related functions
- Added min and max width to chat history style
- Chat auto scrolling when scrolled to bottom
- Fixed chat log of new connected players
- Minor logic fixes

Commit:
Chat changes

- Created a Chat class and gathered chat related functions
- Added min and max width to chat history style
- Chat auto scrolling when scrolled to bottom
- Fixed chat log of new connected players
- Minor logic fixes

aeuu commented on public/scripts/model/Chat.js:
```diff
@@ -0,0 +1,56 @@
+const HTML_classes = {
```
I would suggest renaming this to htmlClasses to preserve consistency,

Commit:
Minor chat changes

Autoscroll when the player sends a message

Commit:
Minor chat changes

Chat text input don't blur after player sends a message

valkyrienyanko reviewed (changes requested):
(1)
Observation: When resizing the chat, you go up in chat a bit.
Expected: When resizing the chat, you should always stay at the position you were looking at (e.g. bottom of chat)

(2) 
Observation: Clicking the chat history highlights it.
Expected: Clicking the chat history does not highlight it.

(3)
Observation: When client A connects everything works. When client B connects, client A does not receive a message saying client A connected.
Expected: See a message from client A when they connect from client B.

Commit:
Minor chat changes

- Added jQuery
- Scroll to bottom at chat window resize
- Removed chat history highlight when focused
- Fixes variable typo
